src/component/mxgraph/GraphCellUpdater.ts

```diff
@@ -66,33 +66,33 @@ export default class GraphCellUpdater {
   }
 
   addOverlays(bpmnElementId: string, overlays: Overlay | Overlay[]): void {
-    const mxCell = this.graph.getModel().getCell(bpmnElementId);
-    if (!mxCell) {
+    const cell = this.graph.getModel().getCell(bpmnElementId);
+    if (!cell) {
       return;
     }
     ensureIsArray(overlays).forEach(overlay => {
       const bpmnOverlay = new MxGraphCustomOverlay(overlay.label, this.overlayConverter.convert(overlay));
-      this.graph.addCellOverlay(mxCell, bpmnOverlay);
+      this.graph.addCellOverlay(cell, bpmnOverlay);
     });
   }
 
   removeAllOverlays(bpmnElementId: string): void {
-    const mxCell = this.graph.getModel().getCell(bpmnElementId);
-    if (!mxCell) {
+    const cell = this.graph.getModel().getCell(bpmnElementId);
+    if (!cell) {
       return;
     }
-    this.graph.removeCellOverlays(mxCell);
+    this.graph.removeCellOverlays(cell);
   }
 
   updateStyle(bpmnElementIds: string | string[], styleUpdate: StyleUpdate): void {
     if (!styleUpdate) {
-      // We don't want to create an empty transaction and verify if there are cells with id include in bpmnElementIds
+      // We don't want to create an empty transaction and verify if there are cells with id include in bpmnElementIds.
+      // This could be improved by also stopping processing if styleUpdate has no relevant properties.
       return;
     }
 
-    // In the future, this method can be optimized by not processing if styleUpdate has no relevant properties defined.
     const model = this.graph.getModel();
-    const cells = ensureIsArray<string>(bpmnElementIds)
+    const cells = addCellIdsOfMessageFlowIcons(ensureIsArray<string>(bpmnElementIds))
```

Comment:
**suggestion:**  
I think the reading is a bit complicated. Instead, could we consider using the following code snippet for better clarity ?

```typescript
const cells = [...ensureIsArray<string>(bpmnElementIds), ...getCellIdsOfMessageFlowIcons()];

function getCellIdsOfMessageFlowIcons(): string[] { // Since the message flow icon is an array.
  // Since the message flow icon is stored in a specific cell, it's important to keep it in sync.
  return cellIds.map(id => messageFlowIconId(id))
}
```

Reply:
About the name, `withCellIdsOfMessageFlowIcons` will be more appropriate. The function returns the original array augmented by the cell identifiers of the message flow icon.
This is a pattern already used in the configuration (rollup) and use in a lot of lib.

In addition, `withCellIdsOfMessageFlowIcons` could take a `string | string[]` parameter and handle the call to `ensureIsArray` to make the code easier to read. This will make the symmetry between the 2 calls more explicit.
```ts
const cells = withCellIdsOfMessageFlowIcons(bpmnElementIds)
...
function withCellIdsOfMessageFlowIcons(bpmnElementIds: string | string[]): string[] {
  const cellIds = ensureIsArray<string>(bpmnElementIds);
  // The message flow icon is stored in a dedicated Cell, so it must be kept in sync
  return cellIds.concat(...cellIds.map(id => messageFlowIconId(id)));
}
```



The proposed implementation is the result of a refactoring to improve the intention and remove duplication.
The processing appeared twice: one for the style update, one for the style reset
```ts
const cellIds = ensureIsArray<string>(bpmnElementIds);
cellIds.push(...cellIds.map(id => messageFlowIconId(id)));
```

What is suggested in this comment doesn't  work. The ids for message flow icon require to pass the array of original ids.
Currently, the function doesn't have argument.
This could be something like the following but the processing will be duplicated, this won't be clearer because of a lot of technical code

```ts
const cells = [...ensureIsArray<string>(bpmnElementIds), ...getCellIdsOfMessageFlowIcons(ensureIsArray<string>(bpmnElementIds))]
...
// OR
const cells = ensureIsArray<string>(bpmnElementIds).concat(...getCellIdsOfMessageFlowIcons(ensureIsArray<string>(bpmnElementIds))
...
```
Otherwise, this must be done in two steps, but the duplication is still there.
```ts
const cellIds = ensureIsArray<string>(bpmnElementIds);
const cells = cellIds.concat(...getCellIdsOfMessageFlowIcons(cellIds)
...
```